library/alloc/src/vec/into_iter.rs

```diff
@@ -267,7 +266,7 @@ impl<T, A: Allocator> DoubleEndedIterator for IntoIter<T, A> {
             None
         } else if mem::size_of::<T>() == 0 {
             // See above for why 'ptr.offset' isn't used
-            self.end = unsafe { arith_offset(self.end as *const i8, -1) as *mut T };
+            self.end = self.ptr.wrapping_byte_sub(1);
```

Comment:
Is this right? It's using `self.ptr` instead of `self.end`

Reply:
Indeed this is wrong; should be fixed by https://github.com/rust-lang/rust/pull/101237.

Reply:
I'm sorry 🥲